### DISCUSSION TITLE
Hiking and mtb relation routes

### DISCUSSION THREAD

**Discussion by damianofalcioni** [organicmaps/organicmaps#865] [Ideas]
A layer with coloured and named hiking and mtb relation routes will be great.
An (I think) great sample is provided by mapy.cz in their outdoor map layer: [https://en.mapy.cz/turisticka](https://en.mapy.cz/turisticka)

**Comment by zorglubu** [comment]
Would be really great. Another example is here : https://hiking.waymarkedtrails.org/#route?id=13098375&map=14!46.2998!6.7721

**Reply by biodranik** [reply]
Are you speaking about OSM data or about routes created somewhere else? On the last link there is a KML download button, so the route may be easily imported now into Organic Maps.

**Reply by pred2k** [reply]
Whats so great with hiking.waymarkedtrails.org and mapy.cz is, that they highlight hiking trails and show the name of the trails, like here at Mount Snowdon
https://de.mapy.cz/turisticka?x=-4.0925496&y=53.0755623&z=14
https://hiking.waymarkedtrails.org/#route?id=13098375&map=14.0/53.0669/-4.0641

No need to download/import a KML files.

**Comment by zorglubu** [comment]
I am speaking about relations created in OSM, like this one : https://www.openstreetmap.org/relation/13096372
and the ability to follow it offline. Thank you.

**Reply by biodranik** [reply]
Cool, then it's feasible. Which UI do you think would be easy for users? I can imagine a layer button which highlights some routes.

But what about the following mode? Would it be enough to just show your position on top of the map and the route, like now?

**Comment by zorglubu** [comment]
First of all, let's keep organicmaps as light and easy to use as it is today, for normal users ! Those who just want to go from one point to another one.

Extra functionalities like this one should be in expert mode, allowing to display either hiking or vtt routes, elevation and contour lines.
Showing the route and where we are should be OK in my mind. But often, there are several overlapping routes, then it would be nice to : 
- click on a route
- select one in a list if there are different routes
- show the details (see below)
- be able to mark this route in order to keep it highlighted, even if the phone has been shut down.

For an example of what I mean, click on a trail on https://hiking.waymarkedtrails.org/#?map=17!46.3125!6.7824

**Comment by cyberchrime** [comment]
Somehow similar to the link provided by @zorglubu is the UI of https://www.cyclosm.org, which is solely based on OSM data. It is especially designed for cyclists and is capable of rendering bike routes and tracks, but also displays speed limited streets in different colors when zooming in.This makes it quite easy to find routes which are safe to use for cyclists. It further displays bike shops, water taps, etc.

**Comment by nonobio** [comment]
Hi :)
I hadn't seen this discussion, that's also what I was wondering: https://github.com/orgs/organicmaps/discussions/8233

**Reply by eblis** [reply]
I guess this is an old issue, I can see there's a Hiking layer available in Organic Maps now, which shows the hiking paths.
A small question I have, not sure where to best ask .. would it be possible to show the trail symbol on the map as well, like for example the way Mapy does it (the circle, line, C symbols, etc)
<img width="1121" height="745" alt="image" src="https://github.com/user-attachments/assets/c3e738bb-a3e3-42b2-9760-d3279440df17" />
<img width="1077" height="546" alt="image" src="https://github.com/user-attachments/assets/abf05418-5b89-42a3-a8d6-2d19e024e58c" />

**Reply by biodranik** [reply]
Is this trail symbol present in OpenStreetMap.org map data?

**Comment by eblis** [comment]
Yes, it's available as `osmc:symbol` or `symbol`
<img width="1479" height="709" alt="image" src="https://github.com/user-attachments/assets/af38d3b5-b958-4b07-8b98-3624a49c9006" />
<img width="1535" height="836" alt="image" src="https://github.com/user-attachments/assets/eda2abbb-f6e7-42ec-acbb-76bdb7f1b779" />

**Reply by biodranik** [reply]
Is there a list of SVG icons for these symbols?

**Reply by eblis** [reply]
Doesn't seem to be SVGs, but there are [PNGs available here](https://wiki.openstreetmap.org/wiki/Trail_markings)

**Reply by eblis** [reply]
I've also written about this feature request here, #7500 which seems to be related to adding shields which i'm guessing is the same as the symbols I mentioned.

**Reply by biodranik** [reply]
A good start would be:
1. Get existing stats on all possible values of a `symbol` tag and their count in OSM.
2. Prepare a list of SVG icons for at least the top values (ideally for all values).

Then it will be easy to integrate. @euf what are the recommended dimensions for SVGs? 24x24?

**Reply by eblis** [reply]
Some helpful information:
 - [Symbols for marked hiking routes](https://www.wanderreitkarte.de/symbols_en.html)
 - [osmc:symbol help](https://wiki.openstreetmap.org/wiki/Key:osmc:symbol?uselang=en)

I've extracted all `osmc:symbol` values using Overpass
[osmc_symbol_counts.zip](https://github.com/user-attachments/files/22171212/osmc_symbol_counts.zip)

**Reply by eblis** [reply]
I started working on a simple python script to get all the PNGs from the link I shared and convert them to SVGs, but quickly ran into problems.

Using the table with the PNGs isn't going to give us the correct (or rather, all) symbols.
There are quite a few symbols that are not present on that page, but are present on trails, for example this list: https://wiki.openstreetmap.org/wiki/Lokale_Wanderwege_im_Landkreis_Amberg-Sulzbach

A different way of getting the symbols is probably needed, maybe we can iterate over all known trails and grab their symbols from a page like the [one above](https://wiki.openstreetmap.org/wiki/Lokale_Wanderwege_im_Landkreis_Amberg-Sulzbach).

Also, can Organic Maps show PNGs or does it have to be SVGs ?

**Reply by biodranik** [reply]
Should be SVGs to display properly on different devices and resolutions.

**Reply by eblis** [reply]
Is it possible to combine different SVGs into one image ?
I.e. have a SVG for hiker then color it green, red, blue, whatever. Have a 3 striped colored SVG, color bottom and top stripes one color, set a different color for the middle one, basically create the final image at runtime based on all the tags in in `osmc:symbol` ?

`way-color :background :foreground :foreground2 :text:text-color`

**Reply by biodranik** [reply]
Yes, in theory. How will it look?

**Reply by eblis** [reply]
We would have SVGs with the various groups that you can have, like line, cross, diagonal, etc and we adjust the colors programatically.
For example for `green:green:yellow_bar` we show <img width="16" height="16" alt="image" src="https://github.com/user-attachments/assets/4a9b127e-92da-4b2c-bc8a-ab1a1675cc40" /> which is using the base <img width="16" height="16" alt="image" src="https://github.com/user-attachments/assets/8aa7daba-6ae9-49ca-9486-977ddfbb3141" /> but uses green for the background and yellow for the bar instead of red and black.

We also have a base image for `hiker` <img width="16" height="16" alt="image" src="https://github.com/user-attachments/assets/adad1580-1eb6-4ff1-af2f-332139a94f7a" /> which we can add to the center of shields, so we can render stuff like `red:red_diamond_line:yellow_hiker`  <img width="22" height="16" alt="image" src="https://github.com/user-attachments/assets/82876507-6f1b-4a70-b1c4-191534252c85" />

A list of groups can be [found here ](https://hiking.waymarkedtrails.org/osmc_symbols.html) I think

**Reply by biodranik** [reply]
So the first step is to get the SVG rendering library in; there was a WIP PR for it.

**Reply by eblis** [reply]
Is there any help needed for that PR or is it mostly done ?
